feat: Add JSON log format support for Datadog integration

rcdexta opened the pull request:
This PR adds a `--log-format` parameter to gizmosql_server that allows switching between text (default) and JSON log formats. The JSON format is optimized for ingestion by log aggregation services like Datadog.

## Changes

- Added `--log-format` parameter to gizmosql_server CLI with options: text (default), json
- Updated all logging statements to output JSON when format is set to "json"
- Maintained backward compatibility with default text format
- Applied consistent logging across both DuckDB and SQLite backends

## JSON Log Format

When using `--log-format json`, all logs are output as single-line JSON objects with:
- `level`: Log level (INFO, WARNING, ERROR)
- `message`: Human-readable message
- Additional context fields like `query`, `component`, `version`, `engine`, etc.

## Example Usage

```bash
# Default text format (backward compatible)
gizmosql_server --database-filename data.db --print-queries

# JSON format for Datadog
gizmosql_server --database-filename data.db --print-queries --log-format json
```

## Testing

- Verified both text and JSON formats work correctly
- All SQL operations tested and working
- Query logging properly outputs in JSON format when enabled
- Backward compatibility maintained

prakashbalaji commented on src/duckdb/duckdb_server.cpp:
```diff
@@ -410,9 +424,13 @@ class DuckDBFlightSqlServer::Impl {
-                << request.query << ";\n"
-                << std::endl;
+      if (log_format_ == "json") {
+        std::cout << R"({"level":"INFO","message":"Client running SQL command: )"
```
Statements with ARROW_LOG(INFO) has checks for log level enabled before emitting the logs. The json logs have level inside the json and not sure if it will be easy to implement something like log filtering to enable only info logs and above or error logs and above etc. 
```
bool ArrowLog::IsLevelEnabled(ArrowLogLevel log_level) {
  return log_level >= severity_threshold_;
}
```

prakashbalaji commented on src/duckdb/duckdb_server.cpp:
```diff
@@ -318,6 +321,17 @@ class DuckDBFlightSqlServer::Impl {
+    
+    if (print_queries_) {
+      if (log_format_ == "json") {
+        std::cout << R"({"level":"INFO","message":"Client executing SQL statement: )"
```
Should the json logging be wrapped in a function call to aid with level check etc ?

prakashbalaji commented on src/library/gizmosql_library.cpp:
```diff
@@ -96,7 +143,12 @@ arrow::Result<std::shared_ptr<flight::sql::FlightSqlServerBase>> FlightSQLServer
+      std::cout << R"({"level":"INFO","message":"Using mTLS CA certificate: )" 
+                << mtls_ca_cert_path.string() << R"("})" << std::endl;
+    } else {
+      std::cout << "Using mTLS CA certificate: " << mtls_ca_cert_path << std::endl;
```
should be ARROW_LOG(INFO)

rcdexta commented:
1. Updated JSON log format to match the required Datadog structure:
    - Added @timestamp field with ISO8601 format including nanoseconds
    - Added @version field (set to "1")
    - Consolidated all metadata into the message field
    - Added logger_name field (differentiates between "gizmosql_server" and "duckdb_server")
    - Added thread_name field with thread ID
    - Kept level field for log severity
  2. Addressed log level filtering concern:
    - Added log level checking using arrow::util::ArrowLog::IsLevelEnabled()
    - JSON logs now respect the Arrow log level settings (controlled by ARROW_DEFAULT_LOG_LEVEL environment variable)
    - Provides consistent behavior between JSON and regular logging modes
  3. Created centralized LogMessage functions:
    - One in gizmosql_library.cpp for the main server
    - One in duckdb_server.cpp for DuckDB-specific logs
    - Both functions handle level checking, formatting, and output
  4. Example output:
  ```json
  {"@timestamp":"2025-08-14T05:36:48.660581000Z","@version":"1","message":"GizmoSQL starting [copyright_year=2025, license=Apache License 2.0]","logger_name":"gizmosql_server","thread_name":"thread-0x1ee5c20c0","level":"INFO"}
  {"@timestamp":"2025-08-14T05:37:13.860435000Z","@version":"1","message":"Client executing SQL statement: SELECT 'JSON format test' as status [component=duckdb_server]","logger_name":"duckdb_server","thread_name":"thread-0x16e823000","level":"INFO"}
```
